expense-splitter/src/components/expense/ExpenseDetail.jsx

```diff
@@ -1,27 +1,135 @@
+
+  // get total amount of expense
+  const totalAmount = expenseDetails.amount;
+
+  // get group connected to expense
+  const expenseGroup = groupData.filter(
+    (group) => group.id === expenseDetails.groupId,
+  );
+
+  // turn array into single object
+  const findGroup = expenseGroup.find(
+    (group) => group.id === expenseDetails.groupId,
+  );
+
+  // get friends names from goup
+  const getFriends = friends.filter((friend) =>
+    findGroup.friendIDs.includes(friend.id),
+  );
+
+  // get weight array from expense object
+  const getWeight = expenseDetails.weight;
+
+  // set data for pie chart to be array of contribution values
+  const pieChartData = getWeight.map((item) =>
+    ((item.percentage / 100) * totalAmount).toFixed(2),
+  );
+
+  // sort payers by contribution amount
+  const sortedContributions = getWeight.sort(function (a, b) {
+    return b.percentage - a.percentage;
+  });
+
+  // create PieChart function
+  function PieChart() {
+    const chartRef = useRef(null);
+
+    useEffect(() => {
+      const dataPie = {
+        datasets: [
+          {
+            // label: getFriends.map(item => item.name),
+            data: pieChartData,
+            backgroundColor: [
+              "#05299e",
+              "#761699",
+              "#af008b",
+              "#d80076",
+              "#f4235d",
+              "#ff5344",
+              "#ffa600",
+            ],
+            hoverOffset: 4,
+          },
+        ],
+      };
+
+      const configPie = {
+        type: "pie",
+        data: dataPie,
+        options: {},
+      };
+
+      const pieChart = new Chart(chartRef.current, configPie);
+
+      return () => {
+        pieChart.destroy();
+      };
+    });
+
+    return (
+      <div className="mx-auto max-w-sm overflow-hidden lg:max-w-lg">
+        <canvas className="p-4" ref={chartRef}></canvas>
+      </div>
+    );
+  }
+
+  const memberDisplay = sortedContributions.map((item) => {
+    return (
+      <Card
+        key={item.id}
+        id={item.id}
+        type={"friend"}
+        icon={"fa-user"}
+        title={getFriends.find((i) => i.id === item.friendId).name}
+        subtitle={item.percentage + "%"}
+        price={((item.percentage / 100) * totalAmount).toFixed(2)}
+      />
+    );
+  });
 
   return (
     <div>
-      <h1>{singleExpense.name}</h1>
-      <p>{singleExpense.category}</p>
-      <p>{singleExpense.description}</p>
-      <p>{singleExpense.amount}</p>
-      <Button
-        variant={"small"}
-        onClick={() => navigate("/expenses")}
-        className={"bg-accent"}
-      >
-        Back
-      </Button>
+      <h1 className="border-b-2 border-[#05299e] p-2 text-center">
+        {expenseDetails.name}
+      </h1>
+      <h1 className="mb-6 border-b-2 border-[#05299e] p-2 text-center">
+        ${expenseDetails.amount}
+      </h1>
+      <p className="mb-3 font-bold">
+        Info: <p>{expenseDetails.description}</p>
```

Comment:
<p/ > elements should not be nested within another <p/ > element

Reply:
I did this because I wanted Info to be bold, but not the description.

Not sure how to do otherwise.

Reply:
I think a span would do it

Reply:
working on this now